examples/demo.js

```diff
@@ -12,21 +12,24 @@ const format = workbook.addFormat();
 /* Set the bold property for the format */
 format.setBold();
 
-/* Change the column width for clarity. */
-worksheet.setColumn(0, 0, 20);
-
 /* Write some simple text. */
 worksheet.writeString(0, 0, "Hello");
 
 /* Text with formatting. */
-worksheet.writeString(1, 0, "World", format);
+worksheet.writeString(0, 1, "World", format);
 
 /* Write some numbers. */
-worksheet.writeNumber(2, 0, 123);
-worksheet.writeNumber(3, 0, 123.456);
+worksheet.writeNumber(1, 1, 123);
+worksheet.writeNumber(2, 1, 123.456);
+
+/* Change the column width for clarity. */
+worksheet.setColumn(1, 1, 20);
+
+/* Apply autofilters */
+worksheet.autofilter(0, 1, 2, 1);
```

Comment:
The examples are intended to produce the exact output as their C equivalent so we can compare them. You'll have to check out `examples/autofilter.c` and create a JS version. It looks like we might need an additional `worksheet.filterColumn` API for that (we can combine `worksheet_filter_column` and `worksheet_filter_column2` into one method with two overloads). I can do that, or you can add it to this PR.

Reply:
Would be nice if you could do it, writing in c, c++ and js and doing my best not to leak memory is giving me brain damage little by little :) Although filter api seems simpler than data validation

Reply:
No worries. For these things it's best to run with the sanitizer while debugging, which makes it much easier to catch any memory issues. You can do this with `node-gyp rebuild --asan=1 --ubsan=1`.